pkg/rpc/server/server.go

```diff
@@ -69,14 +79,39 @@ func (s *Server) Start(errChan chan error) {
 	s.Handler = http.HandlerFunc(s.requestHandler)
 	s.log.Info("starting rpc-server", zap.String("endpoint", s.Addr))
 
-	errChan <- s.ListenAndServe()
+	if cfg := s.config.TLSConfig; cfg.Enabled {
+		s.https.Handler = http.HandlerFunc(s.requestHandler)
+		s.log.Info("starting rpc-server (https)", zap.String("endpoint", s.https.Addr))
+		go func() {
+			err := s.https.ListenAndServeTLS(cfg.CertFile, cfg.KeyFile)
+			if err != nil {
+				s.log.Error("error in listenTLS", zap.Error(err))
```

Comment:
`failed to start TLS RPC server`

Reply:
I mean these `listen/listenTLS` are not immediately obvious to user. And it's important to mention RPC.